docs/DeveloperGuide.md

```diff
@@ -251,74 +278,83 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+new classes such as `EventName`, `EventDate` and `EventTime` were also implemented. These classes function as abstractions 
+for the name, date and time of an `Event` respectively. These abstractions are helpful in ensuring reliability of 
+the program by hiding the underlying details.
 
+### Delete Event feature - deleteEvent
 
-### Remove Event feature
-
-This feature allows Managera users to remove an existing event at a particular index of the displayed list.
+This feature allows Managera users to delete an existing event at a particular index of the displayed event list.
 
 #### Implementation Details
 
-The `AddressBookParser` is responsible for determining the type of `Command` to be created from user input,
-we can simply add a new `commandType` case for `RemoveEventCommand` in `AddressBookParser`.
+Since the `AddressBookParser` is responsible for determining the type of `Command` to be created from user input,
+we can simply add a new `commandType` case for `DeleteEventCommand` in `AddressBookParser`.
 
-The `RemoveEventCommandParser` reads the user's input and passes it to `ParserUtil` which returns an `Index`. This 
-`Index` will be supplied to the `removeEventCommand` to be executed. 
+The `DeleteEventCommandParser` reads the user's input and passes it to `ParserUtil` which returns an `Index`.
+If the index given by the user is not a zero-based index, a `ParseException` will be thrown before
+the parser creates the command to prevent any further error. If the index is valid, a `DeleteEventCommand` 
+will be created by the parser with the created `Index`.
 
-When the command is executed, there is a check to ensure the index is within the range of the displayed list. 
+When the command is executed, there is a check to ensure the index is within the range of the displayed list. If the 
+index is out of range, a `CommandException` is thrown, and an error message is displayed to the user. Otherwise, the 
+execution will continue, and the respective Event at the index is deleted.
 
 #### Design Considerations:
-##### Aspect: Specifying Event to be Removed: 
+##### Aspect: Specifying event to be deleted: 
 
-* **Alternative 1 (Current Choice)**: Remove by Index:
+* **Alternative 1 (Current Choice)**: Delete by index:
     * Pros:
         1. The index of the event in the currently displayed list can be directly used.
         2. Faster and more intuitive for users.
     * Cons:
         1. The index is independent of the Event, which may lead to an inconsistent user experience.
          
-* **Alternative 2**: Remove by Event Name:
+* **Alternative 2**: Delete by event name:
     * Pros:
         1. User can be more sure of the event they are removing, since it is referencing the Event name.
     * Cons:
-        1. The user has to fully match the Event name, which is much more cumbersome. 
+        1. The user has to fully match the Event name, which is much more cumbersome.
 
 ### Filter Event feature
```

Comment:
Should we be consistent and add the "command word" e.g. `filterEvents` in this case? Since for addEvent and deleteEvent they are added

Reply:
👌